app.rb

```diff
@@ -0,0 +1,113 @@
+require './book'
+require './rental'
+require './person'
+require './student'
+require './teacher'
+
+class App
+  attr_accessor :books, :rentals, :people, :student, :teacher
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    person_type = person_type_input
+    if person_type == 1
+      person = create_student
+    elsif person_type == 2
+      person = create_teacher
+    else
+      puts 'Invalid input.'
+      return
+    end
+    push_person_to_list(person)
+    puts 'Person created successfully'
+  end
+
```

Comment:
- [ ] I could not create a person (student or teacher). The app crashes every time I try to do that. I was only able to create books. Kindly fix that.